src/muc_light/mod_muc_light_api.erl

```diff
@@ -0,0 +1,320 @@
+    {ok, iolist()}.
+remove_user_from_room(RoomJID, SenderJID, RecipientJID) ->
+    ok = change_affiliation(RoomJID, SenderJID, RecipientJID, <<"none">>),
+    {ok, io_lib:format("User ~s kicked successfully", [jid:to_binary(RecipientJID)])}.
+
+-spec send_message(jid:jid(), jid:jid(), binary()) ->
+    {ok | not_room_member | muc_server_not_found, iolist()}.
+send_message(#jid{lserver = MUCServer} = RoomJID, SenderJID, Message) ->
+    case mongoose_domain_api:get_subdomain_host_type(MUCServer) of
+        {ok, HostType} ->
+            Body = #xmlel{name = <<"body">>,
+                          children = [ #xmlcdata{ content = Message } ]
+                         },
+            Stanza = #xmlel{name = <<"message">>,
+                            attrs = [{<<"type">>, <<"groupchat">>}],
+                            children = [ Body ]
+                           },
+            SenderBare = jid:to_bare(SenderJID),
+            case is_user_room_member(HostType, jid:to_lus(SenderBare), jid:to_lus(RoomJID)) of
+                true ->
+                    RoomBare = jid:to_bare(RoomJID),
+                    ejabberd_router:route(SenderBare, RoomBare, Stanza),
+                    {ok, "Message sent successfully"};
+                false ->
+                    ?USER_NOT_ROOM_MEMBER_RESULT
+            end;
+        {error, not_found}->
+            ?MUC_SERVER_NOT_FOUND_RESULT
+    end.
+
+-spec delete_room(jid:jid(), jid:jid()) ->
+    {ok | not_allowed | room_not_found | muc_server_not_found , iolist()}.
+delete_room(#jid{lserver = MUCServer} = RoomJID, UserJID) ->
+    case mongoose_domain_api:get_subdomain_host_type(MUCServer) of
+        {ok, HostType} ->
+            case get_room_user_aff(HostType, RoomJID, UserJID) of
+                {ok, owner} ->
+                    ok = mod_muc_light_db_backend:destroy_room(HostType, jid:to_lus(RoomJID)),
+                    ?ROOM_DELETED_SUCC_RESULT;
+                {ok, none} ->
+                    ?USER_NOT_ROOM_MEMBER_RESULT;
+                {ok, member} ->
+                    {not_allowed, "Given user cannot delete this room"};
+                {error, room_not_found} ->
+                    {room_not_found, "Cannot remove not existing room"}
+            end;
+        {error, not_found}->
+            ?MUC_SERVER_NOT_FOUND_RESULT
+    end.
+
+-spec delete_room(jid:jid()) -> {ok | room_not_found | muc_server_not_found, iolist()}.
+delete_room(RoomJID) ->
+    try mod_muc_light:delete_room(jid:to_lus(RoomJID)) of
+        ok ->
+            ?ROOM_DELETED_SUCC_RESULT;
+        {error, not_exists} ->
+            ?ROOM_NOT_FOUND_RESULT
+    catch
+        error:{muc_host_to_host_type_failed, _, _} ->
+            ?MUC_SERVER_NOT_FOUND_RESULT
+    end.
+
+-spec get_room_messages(jid:jid(), integer() | undefined,
+                        mod_mam:unix_timestamp() | undefined) ->
+    {ok, list()} | {muc_server_not_found | internal, iolist()}.
```

Comment:
Maybe it would be possible to extend this `list` type to make it easier to understand how the messages are structured?